libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/limitValues/shared/limitingItemsHook.ts

```diff
@@ -308,18 +337,59 @@ const getDatasetTitle = (
     dateDataSet: ObjRef | undefined,
     defaultDateFilterTitle: string,
     filterConfigByDimension: IDashboardDateFilterConfigItem[],
-    intl: IntlShape,
     filterConfig?: IDashboardDateFilterConfig,
 ) => {
     const usedConfig = dateDataSet
         ? filterConfigByDimension.find((config) => areObjRefsEqual(config.dateDataSet, dateDataSet))?.config
         : filterConfig;
 
-    const dataSetTitle =
-        !usedConfig || usedConfig?.filterName === "" ? defaultDateFilterTitle : usedConfig?.filterName;
+    return !usedConfig || usedConfig?.filterName === "" ? defaultDateFilterTitle : usedConfig?.filterName;
+};
+
+export const useCommonDateItems = (
+    availableDatasets: ICatalogDateDataset[],
+    dependentCommonDateFilter: IDashboardDateFilter,
+    dependentDateFilters?: IDashboardDependentDateFilter[],
+): IValuesLimitingItemWithTitle[] => {
+    return useMemo(() => {
+        return availableDatasets
+            .filter((availableDataset) => {
+                if (dependentDateFilters) {
+                    return !dependentDateFilters.find(
+                        (item) =>
+                            areObjRefsEqual(availableDataset.dataSet.ref, item.dataSet) &&
+                            item.isCommonDate === true,
+                    );
+                } else {
+                    return false;
+                }
+            })
+            .map((dateDataSet) => {
+                return {
+                    title: dateDataSet.dataSet.title,
+                    item: parseCommonDateFilter(dependentCommonDateFilter, dateDataSet.dataSet.ref),
+                    // TODO: LX-160
+                    isDisabled: false,
+                    type: "commonDate",
+                };
+            });
+    }, [availableDatasets, dependentCommonDateFilter, dependentDateFilters]);
+};
+
+const parseCommonDateFilter = (
+    commonDate: IDashboardDateFilter,
+    dataSet?: ObjRef,
+): IDashboardDependentDateFilter => {
+    invariant(!isUriRef(dataSet));
 
-    return intl.formatMessage(
-        { id: "attributesDropdown.valuesLimiting.depedentDateFilterTitle" },
-        { dataSetTitle },
-    );
+    return {
+        from: commonDate.dateFilter.from,
+        to: commonDate.dateFilter.to,
+        granularity: commonDate.dateFilter.granularity,
+        type: commonDate.dateFilter.type,
+        localIdentifier: dataSet?.identifier ?? "commonDate",
```

Comment:
Wouldn't there be a problem if there will be two filters with the same local identifier in case it will not get parsed for two objects or is that something that would never happen?

Reply:
These case would not happen.